hikey_debian.mk

```diff
@@ -0,0 +1,408 @@
+# S_USER (Secure User - TA) are compiled
+################################################################################
+COMPILE_NS_USER   ?= 64
+override COMPILE_NS_KERNEL := 64
+COMPILE_S_USER    ?= 32
+COMPILE_S_KERNEL  ?= 64
+
+# Normal/secure world console UARTs: 3 or 0 [default 3]
+# NOTE: For Debian build only UART 3 works until we have sorted out how to build
+# UEFI correcly.
+CFG_NW_CONSOLE_UART ?= 3
+CFG_SW_CONSOLE_UART ?= 3
+
+# TODO: Figure out how to handle this in a better way, but we need a version
+# number with major and minor for the debian packages.
+#   <major version>.<minor version>-<package revision>
+OPTEE_PKG_VERSION ?= 2.0-1
+
+# IP-address to the HiKey device
+IP ?= 127.0.0.1
+
+################################################################################
+# Includes
+################################################################################
+-include common.mk
+
+################################################################################
+# Mandatory definition to use common.mk
+################################################################################
+ifeq ($(COMPILE_NS_USER),64)
+MULTIARCH			:= aarch64-linux-gnu
+else
+MULTIARCH			:= arm-linux-gnueabihf
+endif
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+ARM_TF_PATH			?= $(ROOT)/arm-trusted-firmware
+ifeq ($(DEBUG),1)
+ARM_TF_BUILD			?= debug
+else
+ARM_TF_BUILD			?= release
+endif
+
+EDK2_PATH 			?= $(ROOT)/edk2
+ifeq ($(DEBUG),1)
+EDK2_BIN 			?= $(EDK2_PATH)/Build/HiKey/DEBUG_GCC49/FV/BL33_AP_UEFI.fd
+EDK2_BUILD			?= DEBUG
+else
+EDK2_BIN 			?= $(EDK2_PATH)/Build/HiKey/RELEASE_GCC49/FV/BL33_AP_UEFI.fd
+EDK2_BUILD			?= RELEASE
+endif
+
+OUT_PATH			?= $(ROOT)/out
+MCUIMAGE_BIN			?= $(EDK2_PATH)/HisiPkg/HiKeyPkg/NonFree/mcuimage.bin
+BOOT_IMG			?= $(OUT_PATH)/boot-fat.uefi.img
+NVME_IMG			?= $(OUT_PATH)/nvme.img
+SYSTEM_IMG			?= $(OUT_PATH)/debian_system.img
+ROOTFS_PATH			?= $(OUT_PATH)/rootfs
+LLOADER_PATH			?= $(ROOT)/l-loader
+PATCHES_PATH			?= $(ROOT)/patches_hikey
+AESPERF_PATH			?= $(ROOT)/aes-perf
+SHAPERF_PATH			?= $(ROOT)/sha-perf
+DEBPKG_PATH			?= $(OUT_PATH)/optee_$(OPTEE_PKG_VERSION)
+DEBPKG_BIN_PATH			?= $(DEBPKG_PATH)/usr/bin
+DEBPKG_LIB_PATH			?= $(DEBPKG_PATH)/usr/lib/$(MULTIARCH)
+DEBPKG_TA_PATH			?= $(DEBPKG_PATH)/lib/optee_armtz
+DEBPKG_CONTROL_PATH		?= $(DEBPKG_PATH)/DEBIAN
+
+################################################################################
+# Targets
+################################################################################
+all: prepare arm-tf linux boot-img lloader system-img nvme deb
+
+clean: arm-tf-clean edk2-clean linux-clean optee-os-clean optee-client-clean xtest-clean boot-img-clean lloader-clean aes-perf-clean sha-perf-clean
+
+cleaner: clean prepare-cleaner linux-cleaner nvme-cleaner system-img cleaner
```

Comment:
Thanks for the catch Victor. This doesn't cause the re-builds, but still this needs to be fixed. I'll fix that when rebasing and applying the tags.